src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -1404,17 +1404,32 @@ private bool CanShowNewFunctionDialogCommand(object parameter)
         /// </summary>
         private void ShowNewPresetStateDialogAndMakePreset(object parameter)
         {
-            //trigger the event to request the display
-            //of the preset name dialogue
-            var args = new PresetsNamePromptEventArgs();
-            this.Model.OnRequestPresetNamePrompt(args);
-            var IDS = DynamoSelection.Instance.Selection.OfType<NodeModel>().Select(x => x.GUID).ToList();
-            if (args.Success)
+            var selectedNodes = GetInputNodes().ToList();
+
+            //If there are NO input nodes then show the error message
+            if (!selectedNodes.Any())
+            {
+                this.Model.OnRequestPresetWarningPrompt();
+            }
+            else
             {
+                //trigger the event to request the display
+                //of the preset name dialogue
+                var args = new PresetsNamePromptEventArgs();
+                this.Model.OnRequestPresetNamePrompt(args);
+
+                //Select only Input nodes for preset
+                var IDS = selectedNodes.Select(x => x.GUID).ToList();
```

Comment:
A style thing: It's strange to name a local variable `IDS` in all caps, this doesn't follow .NET style guides. Shouldn't this be `ids`?